### PR TITLE
PMM-8770 Added safe collectors params

### PR DESCRIPTION
**Pull request by percona-csalguero**
https://jira.percona.com/browse/PMM-8770

### CODE REVIEW

**Review by JiriCtvrtka — APPROVED** [review]

**Review by BupycHuk — DISMISSED** [review]

**Review by BupycHuk — APPROVED** [review]

**Review by JiriCtvrtka — DISMISSED** [review]

File: api/inventorypb/agents.proto [hunk 1]
```diff
@@ -712,6 +712,11 @@ message AddMongoDBExporterRequest {
   string authentication_database = 15;
   // Custom password for exporter endpoint /metrics.
   string agent_password = 16;
+  // Comma separated list of colletions to get stats from. Can use *
```
Comment (artemgavrilov) [inline api/inventorypb/agents.proto:715]:
```suggestion
  // Comma separated list of collections to get stats from. Can use * .
```

File: api/swagger/swagger-dev.json [hunk 2]
```diff
@@ -3853,6 +3853,20 @@
                   "description": "Custom password for exporter endpoint /metrics.",
                   "type": "string",
                   "x-order": 15
+                },
+                "stats_collections": {
+                  "type": "array",
+                  "title": "Comma separated list of colletions to get stats from. Can use *",
```
Comment (ShashankSinha252) [inline api/swagger/swagger-dev.json:3859]: `description` instead of `title` ?

File: api/swagger/swagger-dev.json [hunk 3]
```diff
@@ -3853,6 +3853,20 @@
...
+                  "items": {
+                    "type": "string"
+                  },
+                  "x-order": 16
+                },
+                "collections_limit": {
+                  "type": "integer",
+                  "format": "int32",
+                  "title": "Collections limit. Only get Databases and collection stats if the total number of collections in the server\nis less than this value. 0: no limit",
```
Comment (ShashankSinha252) [inline api/swagger/swagger-dev.json:3868]: `description` instead of `title` ?

File: api/swagger/swagger-dev.json [hunk 4]
```diff
@@ -20813,6 +20827,20 @@
                   "description": "Custom password for exporter endpoint /metrics.",
                   "type": "string",
                   "x-order": 25
+                },
+                "stats_collections": {
+                  "type": "array",
+                  "title": "Comma separated list of colletions to get stats from. Can use *",
```
Comment (ShashankSinha252) [inline api/swagger/swagger-dev.json:20833]: `description` instead of `title` ?

File: api/swagger/swagger-dev.json [hunk 5]
```diff
@@ -20813,6 +20827,20 @@
...
+                  "items": {
+                    "type": "string"
+                  },
+                  "x-order": 26
+                },
+                "collections_limit": {
+                  "type": "integer",
+                  "format": "int32",
+                  "title": "Collections limit. Only get Databases and collection stats if the total number of collections in the server\nis less than this value. 0: no limit",
```
Comment (ShashankSinha252) [inline api/swagger/swagger-dev.json:20842]: `description` instead of `title` ?

File: api/swagger/swagger.json [hunk 6]
```diff
@@ -1881,6 +1881,20 @@
                   "description": "Custom password for exporter endpoint /metrics.",
                   "type": "string",
                   "x-order": 15
+                },
+                "stats_collections": {
+                  "type": "array",
+                  "title": "Comma separated list of colletions to get stats from. Can use *",
```
Comment (ShashankSinha252) [inline api/swagger/swagger.json:1887]: `description` instead of `title` ?

File: api/swagger/swagger.json [hunk 7]
```diff
@@ -1881,6 +1881,20 @@
...
+                  "items": {
+                    "type": "string"
+                  },
+                  "x-order": 16
+                },
+                "collections_limit": {
+                  "type": "integer",
+                  "format": "int32",
+                  "title": "Collections limit. Only get Databases and collection stats if the total number of collections in the server\nis less than this value. 0: no limit",
```
Comment (ShashankSinha252) [inline api/swagger/swagger.json:1896]: `description` instead of `title` ?

File: api/swagger/swagger.json [hunk 8]
```diff
@@ -15196,6 +15210,20 @@
                   "description": "Custom password for exporter endpoint /metrics.",
                   "type": "string",
                   "x-order": 25
+                },
+                "stats_collections": {
+                  "type": "array",
+                  "title": "Comma separated list of colletions to get stats from. Can use *",
```
Comment (ShashankSinha252) [inline api/swagger/swagger.json:15216]: `description` instead of `title` ?

File: api/swagger/swagger.json [hunk 9]
```diff
@@ -15196,6 +15210,20 @@
...
+                  "items": {
+                    "type": "string"
+                  },
+                  "x-order": 26
+                },
+                "collections_limit": {
+                  "type": "integer",
+                  "format": "int32",
+                  "title": "Collections limit. Only get Databases and collection stats if the total number of collections in the server\nis less than this value. 0: no limit",
```
Comment (ShashankSinha252) [inline api/swagger/swagger.json:15225]: `description` instead of `title` ?

**Review by ShashankSinha252 — DISMISSED** [review]

File: api/managementpb/mongodb.proto [hunk 10]
```diff
@@ -87,6 +87,11 @@ message AddMongoDBRequest {
   string authentication_database = 26;
   // Custom password for exporter endpoint /metrics.
   string agent_password = 27;
+  // Comma separated list of colletions to get stats from. Can use *
```
Comment (artemgavrilov) [inline api/managementpb/mongodb.proto:90]:
```suggestion
  // Comma separated list of collections to get stats from. Can use * .
```

File: api/inventorypb/agents.proto [hunk 11]
```diff
@@ -712,6 +712,11 @@ message AddMongoDBExporterRequest {
...
+  repeated string stats_collections = 19;
```
Comment (artemgavrilov) [inline api/inventorypb/agents.proto:716]: Why 19, not 17?

File: api/inventorypb/json/inventorypb.json [hunk 12]
```diff
@@ -424,6 +424,12 @@
                   "type": "string",
                   "x-order": 13
                 },
+                "collections_limit": {
+                  "type": "integer",
+                  "format": "int32",
+                  "title": "Collections limit. Only get Databases and collection stats if the total number of collections in the server\nis less than this value. 0: no limit",
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply api/inventorypb/json/inventorypb.json:430]: it's autogenerated, we can add description later.

File: api/inventorypb/agents.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (percona-csalguero) [reply api/inventorypb/agents.proto:715]: Fixed

**Review by BupycHuk — APPROVED** [review]

**Review by artemgavrilov — APPROVED** [review]

File: api/inventorypb/agents.proto [hunk 13]
```diff
@@ -712,6 +712,11 @@ message AddMongoDBExporterRequest {
   string authentication_database = 15;
   // Custom password for exporter endpoint /metrics.
   string agent_password = 16;
+  // List of colletions to get stats from. Can use *
+  repeated string stats_collections = 17;
+  // Collections limit. Only get Databases and collection stats if the total number of collections in the server
+  // is less than this value. 0: no limit
+  int32 collections_limit = 18;
```
Comment (artemgavrilov) [inline api/inventorypb/agents.proto:719]: Also, it can be `uint32`.

Comment (artemgavrilov) [reply api/inventorypb/agents.proto:719]: @BupycHuk what do you think?

Comment (BupycHuk) [reply api/inventorypb/agents.proto:719]: @percona-csalguero I remember that we had discussion about using -1 for no limit, right? 0 can be used for default limit like we do for MySQL

**Pull request merged**